aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -470,6 +470,18 @@ module aptos_framework::coin {
         let Coin { value: _ } = source_coin;
     }
 
+    /// Same as be merge function but it return zero source coin for gas fee optimize reason if
+    /// you got a situation that you need merge Coin<CoinType> then you also need zero value Coin<CoinType>,
+    /// for example to feed zero coin to an Uniswap V2 type AMM DEX for doing a swap action.
+    public fun merge_and_return_zero_coin<CoinType>(dst_coin: &mut Coin<CoinType>, source_coin: Coin<CoinType>) : Coin<CoinType> {
```

Comment:
Why not just pass a &mut Coin<CoinType> for source_coin?

Reply:
> Why not just pass a &mut Coin for source_coin?

That will add more move instruction like `ReadRef` and `WriteRef` in low level bytecode, by that method may will negate gas optimize effort.